src/client/Microsoft.Identity.Client/Internal/Requests/MsiAuthRequest.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+
+namespace Microsoft.Identity.Client.Internal.Requests
+{
+    internal abstract class MsiAuthRequest : RequestBase
```

Comment:
Please group all SLC files into a single folder somehow. Maybe in Features or in a vanilla folder. 

And use a prefix like `SLC`, because `Credential` is way to overloaded.